src/Core/ScreenSpaceError.js

```diff
@@ -0,0 +1,88 @@
+import * as THREE from 'three';
+
+const v = new THREE.Vector3();
+const m = new THREE.Matrix4();
+const m2 = new THREE.Matrix4();
+const m3 = new THREE.Matrix4();
+
+function compute(vector, matrix, camera, distance, _3d) {
+    const basis = [
+        new THREE.Vector3(vector.x, 0, 0),
+        new THREE.Vector3(0, vector.y, 0),
+    ];
+
+    if (_3d) {
+        basis.push(new THREE.Vector3(0, 0, vector.z));
+    }
+
+    m2.identity();
+    m2.extractRotation(matrix);
+    m3.identity();
+    m3.extractRotation(camera.camera3D.matrixWorldInverse);
+
+    for (const b of basis) {
+        // Apply rotation
+        b.applyMatrix4(m2);
+        // Apply inverse camera rotation
+        b.applyMatrix4(m3);
+        // Move at 'distance' from camera
+        b.z += -distance;
+        // Project on screen
+        b.applyMatrix4(camera.camera3D.projectionMatrix);
+        // cancel z component
+        b.z = 0;
+        b.x = b.x * camera.width * 0.5;
+        b.y = b.y * camera.height * 0.5;
+    }
+
+    const lengthsq = basis.map(b => b.lengthSq());
+    const min = Math.min.apply(Math, lengthsq);
+    return Math.sqrt(min);
+}
+
+function findBox3Distance(camera, box3, matrix) {
+    // TODO: can be cached
+    m.getInverse(matrix);
```

Comment:
`matrix` it's OBB worldMatrix? you would cache in `OBB.worldMatrixInv`?

Reply:
I'd prefer not to cache anything first (hence the TODO node).

I believe we need to rework the `notifyChange()` mechanism to allow more precise updates / cache invalidation.